server/src/main/java/org/apache/iotdb/db/qp/executor/QueryProcessExecutor.java

```diff
@@ -449,6 +457,40 @@ private boolean deleteStorageGroup(DeleteStorageGroupPlan deleteStorageGroupPlan
     }
     return true;
   }
+  
+  private boolean createDeviceTemplate(CreateDeviceTemplatePlan createDeviceTemplatePlan) 
+      throws QueryProcessException {
+    String deviceType = createDeviceTemplatePlan.getDeviceType();
+    List<MeasurementSchema> schemaList = createDeviceTemplatePlan.getSchemaList();
+    mManager.getDeviceTemplates().put(deviceType, schemaList);
+    return true;
+  }
+  
+  private boolean createDevice(CreateDevicePlan createDevicePlan) throws QueryProcessException {
+    String deviceType = createDevicePlan.getDeviceType();
+    String devicePath = createDevicePlan.getDevicePath();
+    try {
+      if (mManager.getDeviceTemplates().get(deviceType) != null) {
+        List<MeasurementSchema> schemaList = mManager.getDeviceTemplates().get(deviceType);
+        for (MeasurementSchema schema : schemaList) {
+          Path path = new Path(devicePath, schema.getMeasurementId());
+          boolean result = mManager.addPathToMTree(path, 
+              schema.getType(), schema.getEncodingType(), 
+              schema.getCompressor(), schema.getProps());
+          if (result) {
+            storageEngine.addTimeSeries(path, schema.getType(), schema.getEncodingType(), 
+                schema.getCompressor(), schema.getProps());
+          }
```

Comment:
If some of the timeseries are already created, the whole procedure will be interrupted by a MetadataException. We should discuss if it is reasonable.